File: NoteTakingApp/app/src/main/java/com/example/notetakingapp/MainActivity.java
package com.example.notetakingapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.widget.Button;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class MainActivity extends AppCompatActivity {


    private Button addButton;
    private FloatingActionButton floatingActionButton;

    private RecyclerView recyclerView;


    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initVars();
    }

    private void initVars() {
        addButton = findViewById(R.id.addBtn);
        floatingActionButton = findViewById(R.id.fab);
        recyclerView = findViewById(R.id.noteRecView);
    }
}

File: Noter/app/src/main/java/com/example/noter/MainActivity.kt
package com.example.noter

import android.app.Dialog
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.LinearLayout
import androidx.recyclerview.widget.RecyclerView
//import com.example.noter.databinding.ActivityMainBinding
import com.google.android.material.floatingactionbutton.FloatingActionButton

class MainActivity : AppCompatActivity() {

    private lateinit var noteSaveButton: Button

    private lateinit var fab: FloatingActionButton
    private lateinit var recView: RecyclerView

    private lateinit var notLinearLayout: LinearLayout


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)



        setContentView(R.layout.activity_main)

        noteSaveButton = findViewById(R.id.addBtn)

        fab = findViewById(R.id.fab)
        recView = findViewById(R.id.noteRecView)
        notLinearLayout = findViewById(R.id.linearNoteAdder)
        noteSaveButton.setOnClickListener {

            showNoteAdder()
        }

        fab.setOnClickListener {
            showNoteAdder()
        }

    }

    private fun showNoteAdder() {
        val dialog = Dialog(this@MainActivity)
        dialog.setContentView(R.layout.add_note_layout)

        val titleEdit: EditText = dialog.findViewById(R.id.editTitle)
        val titleDesc: EditText = dialog.findViewById(R.id.editSubTitle)
        val buttonAdd: Button = dialog.findViewById(R.id.saveButton)

        dialog.show()
    }
}